### ISSUE TITLE
Keyboard Navigation for PWA

### ISSUE THREAD

**Issue by Davinkjohnson** [episphere/connect#1586]
The main PWA dashboard does not work with standard keyboard navigation, creating an accessibility barrier for anyone who prefers or is dependent upon keyboard navigation. Below are the different actions participants should be able to perform.

- Navigate to interactive elements | Tab (forward), Shift + Tab(backward) | The navigation order should be logical and follow the visual flow of the page.
- Activate a link | Enter 
- Activate a button | Enter or Spacebar | Ensure elements with role="button" can be activated with both keys.
- Check/uncheck a checkbox | Spacebar 
- Navigate radio buttons | ↑/↓ or ←/→ | Users should only be able to select one option in a group.
- Interact with dropdown menus | ↑/↓ to navigate, Spacebar to expand, Enter or Esc to select and collapse

**Comment by robertsamm** [comment]
Hi @bransteitterbr, I caught a couple of issues in dev as well as some "nice to have" items with the keyboard navigation. See below, I provided screenshots where possible. If we can't get everything fixed or if some items need to slide to the next release then I suggest prioritizing items 1 and 3 and 8 (if fixable on our side). However, I am tagging @depietrodeanna here to review this list from a UX perspective to see if she sees anything else that is a "need to fix". 

**Enrollment Screens:**

_Screen between consent submission & UP:_
1. These two forms were not accessible via keyboard nav in order to download. When I would attempt to tab over to them they would be skipped over and went straight to the 'previous' button. The download links for the forms on the previous page (on consent screens) that are similar were accessible.
<img width="464" height="302" alt="Image" src="https://github.com/user-attachments/assets/bfa7bc2e-5a6c-4b4a-b24f-1554f1af85c9" />

_User Profile Screen_
2. When entering an address using the USPS API: I could navigate the drop down but when I hit enter to select it would try to submit the entire page (not just select the address). Clicking 'esc' allowed me to select the address and remain on the page though.
<img width="355" height="548" alt="Image" src="https://github.com/user-attachments/assets/4aa11f73-f0a6-40bd-a297-fcbcb233290d" />

_UP Submission Review Page:_ 
3. Not able to use arrow keys to scroll the page to review the entire list. Can only use 'tab' button which jumps to the buttons at the end. I was able to scroll the previous pages using the up down arrow keys.
<img width="320" height="403" alt="Image" src="https://github.com/user-attachments/assets/5a5d399c-6f6a-46d8-9cbb-8f7f00d62021" />

**Messages Screen:** 
4. Not sure if this is keyboard navigation specific, but I noticed that when trying to open the links in the body of the emails they open up on the same page instead of a new tab so it takes the user out of the PWA. It may be helpful to have them open in a separate tab automatically (like we do for other links throughout the PWA) if the user has limited navigation. However, I am aware that sometimes this is browser specific?
<img width="458" height="391" alt="Image" src="https://github.com/user-attachments/assets/bbbb525c-ce6e-461b-8dda-164c55edb4e4" />

**MyProfiles Screens**
_On This Page Links:_ 
5. The links to skip to the specific sections work. However, when you try to tab to the specific fields it takes you back to the top where you left off. E.g. if I clicked enter on 'Sign In Information' it would display the Sign In Info section but when I clicked tab instead of moving to the 'Update Sign In' button, it would jump me back to the 'On This Page' 'Agreements' link. Not sure if this is something that is fixable, but wanted to point out.
<img width="153" height="99" alt="Image" src="https://github.com/user-attachments/assets/ee127985-f533-4eb2-af92-3dbe5e3ec0c4" />

_Mailing Address Section:_ 
no issues, just noting that when selecting a USPS API address it **did** allow me to click 'enter' and it selected the address without automatically submitting the page. This is preferred behavior compared to entering address in the UP where it auto submits (see issue # 2)

_"Jumping" issue with tabbing throughout MyProfiles:_
6. When selecting 'Go Back' after editing phone and email section, the field it is selected on is all the way at the bottom of the page so you have to tab a bunch of times to get back to the section being edited. Again, not sure if fixable but wanted to point out.
7. When I updated address and clicked submit, I also had this issue where it jumped to the bottom of the page. However, when I edited preferred language and clicked submit it did not jump and I was able to move on to the next section without tabbing over a bunch of times. I saw this issue in a couple other places but not everywhere, does not seem to be consistent.

<img width="350" height="266" alt="Image" src="https://github.com/user-attachments/assets/ba8dea70-c252-459c-9cda-6a209b210451" />

_Captcha for changing phone login:_
8. Not sure if this is a Google issue or something fixable on our end, I was not able to use keyboard navigation to select any of the squares besides the bottom right.
<img width="218" height="308" alt="Image" src="https://github.com/user-attachments/assets/52d9a263-2e60-4311-b405-487eb837baae" />

**Reports Dashboard**
9. Similar issue to # 4, I noticed here that the links within the report on the box preview screen open on the same page instead of opening in a new tab. Would be nice if they opened in a new tab (like the links in the consent/HIPAA preview pages do that use the same display).
<img width="448" height="385" alt="Image" src="https://github.com/user-attachments/assets/d43513d9-748a-4257-859e-5f020baaaca2" />

**Comment by robertsamm** [comment]
For documentation purposes on this github issue, I tested the following as discussed on a PC: sign in screens, enrollment screens (including consent and UP) dashboard/cards, individual pages of the dashboard (except for the survey page), MyProfiles page, Language toggle, Messages page, and individual links throughout. The survey page and surveys themselves are out of scope for this github issue as Quest keyboard nav updates are slated for a future release.

**Comment by depietrodeanna** [comment]
Hi @bransteitterbr, I agree with @robertsamm that these are all nice to have items and not immediately needed as priority fixes. Thanks for testing Amelia!

**Comment by bransteitterbr** [comment]
@robertsamm I have merged in a PR that addresses issues 1, 2, 3, 5.

For issue 4 the email content links.  This issue is not keyboard navigation specific but based on the contents of the email message.  If the email uses target="_blank" on the links then it will open in a new window inside the connectApp otherwise it will open in the same page.  I could attempt to scan the links and add the target if it is missing when displaying messages inside the app.  I will let that up to @Davinkjohnson or @depietrodeanna to decide if we want to try and squeeze that in for this release.

I will continue to look at issues 6/7 to see what I can do to improve the "jumpiness" of the UI and try and make sure the focus makes sense.

For the recaptcha issue 8 I have no way of restyling that interface it is completely under the control of google (by design).  I will say in my testing I was able to use it using the keyboard BUT there is no focus indicator for some reason, so it is hard to use but does technically work.

For issue 9 the report content links opening in the same page.  For the physical activity report we generate that and the links in that report do not open in the same page BUT the DHQ HEI Report is provided by a third party as a preconstructed pdf so I don't have an ability to change the target of those links.  I could research to see if the links can be updated via our PDF library that we use but I would say that it falls outside of this project.

**Comment by robertsamm** [comment]
Hi @bransteitterbr, thanks for the update. Your comments all make sense! For 4 regarding the email links, do you think this target="_blank" is something we should be adding to our hyperlinks in the emails we program moving forward if we want the links to open in a new tab/window? Tagging @depietrodeanna for her awareness who programs the notifications.

**Comment by bransteitterbr** [comment]
@robertsamm and @depietrodeanna , It looks like from what I am seeing some of the emails are already using target="_blank" for their links so this might be a case of the older email not using it and the newer ones do.  For our use case I would say that it makes sense to include the target attribute, but I could implement my link scan code to catch any that might be missed.

**Comment by robertsamm** [comment]
Hi @bransteitterbr I retested 1,2,3 and 5 in dev and all looks great!

**Comment by robertsamm** [comment]
Looks good in stage, ready for prod!

**Comment by robertsamm** [comment]
Hi @bransteitterbr I discussed w/ @depietrodeanna  about whether we want to address any of the remaining "nice to have" items to a future github issue. We determined that items 6,7,8, and 9 we can leave as is. I have create a separate issue for item 4: https://github.com/episphere/connect/issues/1637.
